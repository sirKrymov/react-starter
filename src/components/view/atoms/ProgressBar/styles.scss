@import '../../../../styles/configs';

$bar-height: 24px;

.progress-bar {
  align-items: center;
  display: flex;

  &__outer {
    background-color: var(--bgPrimary);
    border-radius: $bar-height / 2;
    position: relative;
    z-index: 1;
    border: 1px solid var(--borderPrimary);
    height: $bar-height;
    width: 100%;
    overflow: hidden;
  }

  &__fill {
    background-color: var(--secondary);
    border-radius: $bar-height / 2;
    transform: translateY(-50%);
    position: absolute;
    z-index: 2;
    height: $bar-height;
    left: -1px;
    top: 50%;
  }

  &__label {
    text-align: right;
    min-width: 50px;
  }
}
