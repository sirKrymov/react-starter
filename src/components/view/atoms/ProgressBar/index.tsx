import React, { ReactElement } from 'react';

import cn from 'classnames';

import { Typography } from '../Typography';

import './styles.scss';

interface IProps {
  className?: string;
  completed: number | string;
}

export function ProgressBar({ completed, className }: IProps): ReactElement {
  const wrappClass = cn(
    {
      'progress-bar': true
    },
    className
  );

  const enhancedCompleted = Math.floor(+completed);

  return (
    <div className={wrappClass}>
      <div className="progress-bar__outer">
        <div
          className="progress-bar__fill"
          style={{ width: `calc(${enhancedCompleted}% + 1px)` }}
        ></div>
      </div>

      <Typography
        className="progress-bar__label"
        variant="subtitle3"
      >{`${Math.floor(enhancedCompleted)}%`}</Typography>
    </div>
  );
}
