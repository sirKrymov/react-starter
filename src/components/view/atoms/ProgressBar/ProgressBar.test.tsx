import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ProgressBar } from '../ProgressBar';

describe('<ProgressBar> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(<ProgressBar completed={10} />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
