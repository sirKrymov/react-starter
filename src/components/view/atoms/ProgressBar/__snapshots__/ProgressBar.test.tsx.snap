// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__outer"
  >
    <div
      className="progress-bar__fill"
      style={
        Object {
          "width": "calc(10% + 1px)",
        }
      }
    />
  </div>
  <Typography
    className="progress-bar__label"
    variant="subtitle3"
  >
    10%
  </Typography>
</div>
`;
